pkg/registry/http.go

```diff
@@ -72,8 +72,13 @@ func (r SchemaRegistry) DownloadSchema(resourceKind, resourceAPIVersion, k8sVers
 			return url, b.([]byte), nil
 		}
 	}
+	req, _ := http.NewRequest("GET", url, nil)
 
-	resp, err := r.c.Get(url)
+	if token, exist := os.LookupEnv("GITHUB_TOKEN"); exist {
```

Comment:
I'm having some thoughts on how this will work when using multiple schema registries. When using the default registry plus one private one, we will be passing a GITHUB_TOKEN for the default registry too - does this work? This also won't work for multiple registries that might require different GITHUB_TOKENS, though I might be ok with that simplification for the sake of UI.
Also imagine you have a GITHUB_TOKEN set because you use Github, and run kubeconform against non-Github schema registries - the Github token will be passed :thinking:

Reply:
For the first question, if you pass the token to github in a non-authenticated it works (and also all of them will fail if you have an invalid token)
As for using it with the token set and non-github http registries, might be an issue tho... How about if I make a new registry, github-specific and derive it from the http one?
To make it work with generic http registries I think we'll need to change the UI, to get the header name for example 🤔

Reply:
Maybe we can make it 2 env vars?
`KUBECONFORM_AUTH_TOKEN` and `KUBECONFORM_AUTH_HEADER`? It will still be limited to usage on every registry, but might work... I don't see a way of making it per-registry without changing the interface, maybe introducing a config file or sequential parameters

Reply:
Oh another option might be creating a private registry struct implementing the registerer, and passing the values like  `-private-schema-location xxx` `-private-schema-header xxx` `-private-schema-auth xxx` (order should be mandatory tho)